packages/docusaurus-plugin-content-docs/src/client/__tests__/docsClientUtils.test.ts

```diff
@@ -184,12 +185,71 @@ describe('docsClientUtils', () => {
     expect(getActiveVersion(data, '/docs/someDoc')?.name).toBe('version2');
 
     expect(getActiveVersion(data, '/docs/version1')?.name).toBe('version1');
-    expect(getActiveVersion(data, '/docs/version1')?.name).toBe('version1');
+    expect(getActiveVersion(data, '/docs/version1/')?.name).toBe('version1');
     expect(getActiveVersion(data, '/docs/version1/someDoc')?.name).toBe(
       'version1',
     );
   });
 
+  it('sortVersionsByPathDepth', () => {
+    function createVersion(name: string, path: string, isLast: boolean) {
+      return {
+        name,
+        label: name,
+        path,
+        isLast,
+        docs: [],
+        mainDocId: '???',
+        draftIds: [],
+      };
+    }
+    const test1 = [
+      createVersion('current', '/docs', false),
+      createVersion('version2', '/docs/version2', true),
+      createVersion('version1', '/docs/version1', false),
+    ];
+    const test2 = [
+      createVersion('current', '/docs/', false),
+      createVersion('version2', '/docs/version2/', true),
+      createVersion('version1', '/docs/version1/', false),
+    ];
+
+    // docs only website
+    const test3 = [
+      createVersion('current', '/', false),
+      createVersion('version2', '/version2', true),
+      createVersion('version1', '/version1', false),
+    ];
+
+    // docs only with trailing slash
+    const test4 = [
+      createVersion('current', '/', false),
+      createVersion('version2', '/version2/', true),
+      createVersion('version1', '/version1/', false),
+    ];
+
+    expect(sortVersionsByPathDepth(test1)).toEqual([
+      test1[2], // version1
+      test1[1], // version2
+      test1[0], // current
+    ]);
```

Comment:
Would be better if the expectaction was "closer" to the test setup, otherwise this test is hard to understand

It's better to create 4 tests rather than to put many tests into one